### ISSUE TITLE
Hex offline not finding packages

### ISSUE THREAD

**Issue by mwilsoninsight** [NixOS/nixpkgs#61159]
## Issue description

In attempting to get Elixir's Phoenix up and running on Nix, I ran into the small issue that the package was out of date- as were some of its dependencies- either way, it will not build out-of-the-box in its currently offered state.
So I started updating/adding to a checkout of `master`.

I'm finding that I have to edit `./mix.exs` such that all of its dependencies are local path entries rather than hex entries. Otherwise the builds fail.

For example, when building `plug_1_8_0`, I get this error for some of its dependencies:
```
Unchecked dependencies for environment prod:
* telemetry (Hex package)
  the dependency is not available, run "mix deps.get"
* mime (Hex package)
  the dependency is not available, run "mix deps.get"
* plug_crypto (Hex package)
  the dependency is not available, run "mix deps.get"
```

### Steps to reproduce

- Clone nixpkgs
- Add entry in beamPackages for `cowlib_2_7_3` (buildErlangMk)
- Update `cowlib` to `cowlib_2_7_3`
- Add entry in beamPackages for `cowboy_2_6_3` (buildErlangMk)
- Update `cowboy` to `cowboy_2_6_3`
- Add entry in beamPackages for `telemetry_0_4_0` (buildRebar3)
- Add entry in beamPackages for `telemetry` and set it to `telemetry_0_4_0`
- Add entry in beamPackages for `mime_1_3_1` (buildMix)
- Update `mime` to `mime_1_3_1`
- Add entry in beamPackages for `plug_crypto_1_0_0` (buildMix)
- Add entry in beamPackages for `plug_crypto` and set it to `plug_crypto_1_0_0`
- Add entry in beamPackages for `plug_1_8_0` (buildMix)
  - making sure the argument set includes `telemetry`, `mime`, and `plug_crypto` (change `cowboy_x_y_z` to `cowboy`)
  - _and_ that the `beamDeps` attribute passed into `buildMix` contains these, as well `cowboy`
- Update `plug` to `plug_1_8_0`
- `nix-env -if "</possibly/relative/path/to>/nixpkgs" -A beamPackages.plug`

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.34, NixOS, 19.03.172231.7b36963e7a7 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `nixos-19.03.172231.7b36963e7a7`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by 4z3** [comment]
This issue is caused by https://github.com/NixOS/nixpkgs/pull/25655.

**Comment by grahamc** [comment]
Closed by #64884.
